src/main/java/com/sourcegraph/webhook/EventSerializer.java

```diff
@@ -0,0 +1,103 @@
+    }
+
+    private static Adapter<PullRequestUpdatedEvent> PullRequestUpdatedEventAdapter = (element, event) -> {
+        element.addProperty("previousTitle", event.getPreviousTitle());
+        element.addProperty("previousDescription", event.getPreviousDescription());
+        element.add("previousTarget", render(event.getPreviousToBranch()));
+    };
+
+    private static Adapter<PullRequestReviewersUpdatedEvent> PullRequestReviewersUpdatedEventAdapter = (element, event) -> {
+        element.add("removedReviewers", render(event.getRemovedReviewers()));
+        element.add("addedReviewers", render(event.getAddedReviewers()));
+    };
+
+    private static Adapter<PullRequestParticipantStatusUpdatedEvent> PullRequestParticipantStatusUpdatedEventAdapter = (element, event) -> {
+        element.add("participant", render(event.getParticipant()));
+        element.addProperty("previousStatus", event.getPreviousStatus().name());
+    };
+
+    private static Adapter<PullRequestCommentEvent> PullRequestCommentEventAdapter = (element, event) -> {
+        element.add("comment", render(event.getComment()));
+        if (event.getParent() != null) {
+            element.addProperty("commentParentId", event.getParent().getId());
+        }
+    };
+
+    private static Adapter<PullRequestCommentEditedEvent> PullRequestCommentEditedEventAdapter = (element, event) -> {
+        PullRequestCommentEventAdapter.apply(element, event);
+        element.addProperty("previousComment", event.getPreviousText());
+    };
+
+    static {
+        adapters.put(PullRequestUpdatedEvent.class, PullRequestUpdatedEventAdapter);
+        adapters.put(PullRequestReviewersUpdatedEvent.class, PullRequestReviewersUpdatedEventAdapter);
+        adapters.put(PullRequestParticipantApprovedEvent.class, PullRequestParticipantStatusUpdatedEventAdapter);
+        adapters.put(PullRequestParticipantUnapprovedEvent.class, PullRequestParticipantStatusUpdatedEventAdapter);
+        adapters.put(PullRequestParticipantReviewedEvent.class, PullRequestParticipantStatusUpdatedEventAdapter);
+        adapters.put(PullRequestCommentAddedEvent.class, PullRequestCommentEventAdapter);
+        adapters.put(PullRequestCommentDeletedEvent.class, PullRequestCommentEventAdapter);
+        adapters.put(PullRequestCommentEditedEvent.class, PullRequestCommentEditedEventAdapter);
+    }
+
+    @Autowired
+    public EventSerializer(@ComponentImport JsonRenderer renderer) {
+        EventSerializer.renderer = renderer;
+    }
+
+    public EventSerializer(String name) {
+        payload = new JsonObject();
+        payload.addProperty("eventKey", name);
+    }
+
+    public JsonObject serialize(ApplicationEvent event) {
+        buildApplicationEvent(event);
+        if (event instanceof PullRequestEvent) {
+            buildPullRequestEvent((PullRequestEvent) event);
+        }
+
+        Adapter adapter = adapters.get(event.getClass());
+        if (adapter == null) {
```

Comment:
I would invert the control flow here because it is clearer and saves one line of code.

```java
if (adapter != null) {
    adapter.apply(payload, event);
}
return payload;
```